windows/CodePush.Net46/UpdateManager.cs

```diff
@@ -0,0 +1,326 @@
+﻿using Newtonsoft.Json;
+using Newtonsoft.Json.Linq;
+using System.IO;
+using System.IO.Compression;
+using System.Threading.Tasks;
+using System.Runtime.CompilerServices;
+using System;
+#if WINDOWS_UWP
+using Windows.Web.Http;
+#else
+using CodePush.Net46.Adapters.Http;
+using PCLStorage;
+#endif
+
+[assembly: InternalsVisibleTo("CodePush.Net46.UnitTest")]
+
+namespace CodePush.ReactNative
+{
+    internal class UpdateManager
+    {
+        #region Internal methods
+
+        internal async Task ClearUpdatesAsync()
+        {
+            await (await GetCodePushFolderAsync().ConfigureAwait(false)).DeleteAsync().ConfigureAwait(false);
+        }
+
+        internal async Task DownloadPackageAsync(JObject updatePackage, string expectedBundleFileName, Progress<HttpProgress> downloadProgress)
+        {
+            // Using its hash, get the folder where the new update will be saved 
+            var codePushFolder = await GetCodePushFolderAsync().ConfigureAwait(false);
+            var newUpdateHash = (string)updatePackage[CodePushConstants.PackageHashKey];
+            var newUpdateFolder = await GetPackageFolderAsync(newUpdateHash, false).ConfigureAwait(false);
+            if (newUpdateFolder != null)
+            {
+                // This removes any stale data in newPackageFolderPath that could have been left
```

Comment:
Seems like _newUpdateFolder_ is correct name. Fixed.
This implementation (including comments :) ) replicates UWP implementation. I've tried to follow the idea not to change existing code as much as possible.  The proposed change make sense but looks like it should be done in both versions (UWP and NET46) that is I believe can be done as separate PR.